trace/config.go

```diff
@@ -25,10 +28,14 @@ type Config struct {
 	IDGenerator internal.IDGenerator
 }
 
+var configWriteMu sync.Mutex
+
 // ApplyConfig applies changes to the global tracing configuration.
 //
 // Fields not provided in the given config are going to be preserved.
 func ApplyConfig(cfg Config) {
+	configWriteMu.Lock()
+	defer configWriteMu.Unlock()
```

Comment:
LGTM.

Maybe add a comment on `config.Load()` always returning a valid Config object, as config is populated by an `init()`

Reply:
@basvanbeek you need to actually Approve the review for it to be mergeable - LGTM is not enough.

Reply:
We didn't discuss me officially reviewing stuff... but I guess I can ;)